File: src/utils/permissions.py
from nextcord import PermissionOverwrite

def create_ticket_permissions(guild, chef_member, customer_member):
    """
    Returns a dictionary of permission overwrites for a private ticket channel
    """
    overwrites = {
        guild.default_role: PermissionOverwrite(view_channel=False),
        chef_member: PermissionOverwrite(view_channel=True, send_messages=True),
        customer_member: PermissionOverwrite(view_channel=True, send_messages=True)
    }
    return overwrites

File: src/utils/permissions.js
import { PermissionFlagsBits } from "discord.js";

// Helpers for channel permission overwrites
export const createTicketPermissions = (chefId, customerId, guildId) => [
  {
    id: chefId,
    allow: [PermissionFlagsBits.ViewChannel, PermissionFlagsBits.SendMessages]
  },
  {
    id: customerId,
    allow: [PermissionFlagsBits.ViewChannel, PermissionFlagsBits.SendMessages]
  },
  {
    id: guildId,
    deny: [PermissionFlagsBits.ViewChannel]
  }
];
